base/bsupport/logprocessingworker.go

```diff
@@ -14,17 +14,20 @@ type LogProcessingWorker struct {
 	deallocator   *base.LogAllocator
 	procCounter   *base.LogProcessCounter
 	transformList []base.LogTransformFunc
-	serializer    base.LogSerializer
-	chunkMaker    base.LogChunkMaker
-	acceptChunk   base.LogChunkAccepter
+	components    []ProcessingWorkerOutputComponentSet
 	lastChunkTime time.Time
 }
 
+type ProcessingWorkerOutputComponentSet struct {
```

Comment:
What do you think of changing this into an `OutputInterface` embedding LogSerializer and LogChunkMaker from a composite type of them, since there aren't naming conflicts in all 3?

e.g. 

```go
type OutputInterface interface {
    base.LogSerializer  // interface
    base.LogChunkMaker  // interface
    AcceptChunk base.LogChunkAccepter  // function
}
```

and would more more sense to LogProcessingWorker as it only needs to know about an output interface, not how the individual functionalities are structured.